### ISSUE TITLE
Tests fail immediately with error:  "[__NSArrayM insertObject:atIndex:]: object cannot be nil"

### ISSUE THREAD

**Issue by patricklynch** [kif-framework/KIF#699]
I followed the installation instructions and double checked the issues in the troubleshooting section, but my test always fails immediately.  In the error message it looks like it cannot find the main window.  One thing also worth noting, when I create the Cocoa Touch Testing Bundle I selected Swift as the language, not Objective-C. And I did create the bridging header which includes `#import <KIF/KIF.h>`.

MY CODE:
```
import UIKit 

class LoginTests: KIFTestCase {
    
    func testSuccessfulLogin() {
        self.tester().waitForViewWithAccessibilityLabel( "Menu Profile" )
        self.tester().waitForTappableViewWithAccessibilityLabel( "Accessory Settings" )
    }
}
```

ERROR MESSAGE
```
<unknown>:0: error: -[UIAutomationTests.LoginTests testSuccessfulLogin] : failed: caught "NSInvalidArgumentException", "*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil"
(
	0   CoreFoundation                      0x0000000114259c65 __exceptionPreprocess + 165
	1   libobjc.A.dylib                     0x0000000113d80bb7 objc_exception_throw + 45
	2   CoreFoundation                      0x00000001141268ca -[__NSArrayM insertObject:atIndex:] + 954
	3   UIAutomationTests                   0x000000012042d811 -[UIApplication(KIFAdditions) windowsWithKeyWindow] + 193
	4   UIAutomationTests                   0x000000012042ca74 -[UIApplication(KIFAdditions) accessibilityElementWithLabel:accessibilityValue:traits:] + 180
	5   UIAutomationTests                   0x000000012042bd1e +[UIAccessibilityElement(KIFAdditions) accessibilityElementWithLabel:value:traits:error:] + 158
	6   UIAutomationTests                   0x000000012042b7e2 +[UIAccessibilityElement(KIFAdditions) accessibilityElement:view:withLabel:value:traits:tappable:error:] + 178
	7   UIAutomationTests                   0x000000012041e6fa __83-[KIFUITestActor waitForAccessibilityElement:view:withLabel:value:traits:tappable:]_block_invoke + 106
	8   UIAutomationTests                   0x000000012041ac37 -[KIFTestActor tryRunningBlock:complete:timeout:error:] + 167
	9   UIAutomationTests                   0x000000012041af49 -[KIFTestActor runBlock:complete:timeout:] + 137
	10  UIAutomationTests                   0x000000012041b054 -[KIFTestActor runBlock:complete:] + 148
	11  UIAutomationTests                   0x000000012041b130 -[KIFTestActor runBlock:] + 64
	12  UIAutomationTests                   0x000000012041e631 -[KIFUITestActor waitForAccessibilityElement:view:withLabel:value:traits:tappable:] + 337
	13  UIAutomationTests                   0x000000012041e461 -[KIFUITestActor waitForViewWithAccessibilityLabel:value:traits:tappable:] + 209
	14  UIAutomationTests                   0x000000012041dfb3 -[KIFUITestActor waitForViewWithAccessibilityLabel:] + 99
	15  UIAutomationTests                   0x00000001204181c7 _TFC17UIAutomationTests10LoginTests19testSuccessfulLoginfS0_FT_T_ + 503
	16  UIAutomationTests                   0x00000001204183d2 _TToFC17UIAutomationTests10LoginTests19testSuccessfulLoginfS0_FT_T_ + 34
	17  CoreFoundation                      0x000000011414fdec __invoking___ + 140
	18  CoreFoundation                      0x000000011414fc42 -[NSInvocation invoke] + 290
	19  XCTest                              0x000000012047917a -[XCTestCase invokeTest] + 253
	20  XCTest                              0x0000000120479379 -[XCTestCase performTest:] + 150
	21  XCTest                              0x0000000120482c35 -[XCTest run] + 260
	22  XCTest                              0x000000012047808b -[XCTestSuite performTest:] + 379
	23  XCTest                              0x0000000120482c35 -[XCTest run] + 260
	24  XCTest                              0x000000012047808b -[XCTestSuite performTest:] + 379
	25  XCTest                              0x0000000120482c35 -[XCTest run] + 260
	26  XCTest                              0x000000012047808b -[XCTestSuite performTest:] + 379
	27  XCTest                              0x0000000120482c35 -[XCTest run] + 260
	28  XCTest                              0x0000000120475129 __25-[XCTestDriver _runSuite]_block_invoke + 56
	29  XCTest                              0x000000012047fedd -[XCTestObservationCenter _observeTestExecutionForBlock:] + 162
	30  XCTest                              0x0000000120475060 -[XCTestDriver _runSuite] + 269
	31  XCTest                              0x0000000120475a8d -[XCTestDriver _checkForTestManager] + 234
	32  XCTest                              0x0000000120485b20 +[XCTestProbe runTests:] + 182
	33  Foundation                          0x00000001102da1e5 __NSFireDelayedPerform + 387
	34  CoreFoundation                      0x00000001141c1174 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
	35  CoreFoundation                      0x00000001141c0d35 __CFRunLoopDoTimer + 1045
	36  CoreFoundation                      0x0000000114182d3d __CFRunLoopRun + 1901
	37  CoreFoundation                      0x0000000114182366 CFRunLoopRunSpecific + 470
	38  GraphicsServices                    0x0000000116bdfa3e GSEventRunModal + 161
	39  UIKit                               0x0000000110d838c0 UIApplicationMain + 1282
	40  victorious                          0x000000010e41857f main + 111
	41  libdyld.dylib                       0x0000000115a6e145 start + 1
	42  ???                                 0x000000000000000a 0x0 + 10
```

**Comment by phatmann** [comment]
Looks like your Test Host app has no key window. Can you run the test host outside the testing environment on the simulator to see what is going on?

**Comment by patricklynch** [comment]
Yep, my bad.  The test host app runs just fine, but I forgot we added a special condition that doesn't load the window when running unit tests, which also got applied to the automation tests.  I changed the condition and now the KIF tests run perfect.  Thanks for the quick reply!

**Comment by whateverDude** [comment]
@patricklynch : I have the same issue with that exception.  Where do I look for the condition/flag that doesn't load the window when running unit tests?

**Comment by phatmann** [comment]
@whateverDude: Look in the `application:didFinishLaunchingWithOptions:` method.

**Comment by mikelupo** [comment]
Hi @phatmann, I think @whateverDude is all set. His issue was that when he created a test target that it was a UI test target and not a Unit test target.

**Comment by patricklynch** [comment]
Yes, it was my own issue, not with KIF.  I conditionally disabled creation of the application's window for another purpose, which caused that exception to throw.

**Comment by cicerohellmann** [comment]
I'm having the same issue and I don't see how to solve it. Can someone point the way a little more clearer?

**Comment by cicerohellmann** [comment]
MY application:didFinishLaunchingWithOptions:

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    self.window = [[UIWindow alloc] initWithFrame:[[UIScreen mainScreen] bounds]];
    
    ViewController *viewController = [[ViewController alloc] init];
    self.window.rootViewController = viewController;
    
    [self.window makeKeyAndVisible];
    
    // always call after makeKeyAndDisplay.
#if TARGET_IPHONE_SIMULATOR
    [[DCIntrospect sharedIntrospector] start];
#endif
    
    return YES;
}

**Comment by mikelupo** [comment]
Did it ever work or are you newly integrating testing and KIF?

**Comment by cicerohellmann** [comment]
newly, never tried before, just the native one and Spectra+Expecta

**Comment by mikelupo** [comment]
Your project isn't create/set up correctly. 

Don't know if this is your issue, but your test target when you create it should be UNIT test and not UI test target. A common mistake.
Here is an example project that I helped put together for another person needing similar support.
https://github.com/sandeeplakshmi/unitautomation

Hope this helps...

**Comment by cicerohellmann** [comment]
Sorry for not responding in time, I deleted everything and rewrited my project the correct way, created the targets from scratch and it seems to have fixed it up. Probably some garbage in my targets setup that I couldn't change or edit.

Thank you for your attention. 
And if someone happen to have the same issue try to delete everything related to the test (the target and the scheme) and create it again!
I also tried A LOT of other solutions to fix it by editing configurations, none of the others on stack overflow or blogs solved for this specific problem. Even the linking ones and the search for framework folder ones. Just delete it!
